src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/ApiSteps.java

```diff
@@ -106,7 +110,15 @@ public void cleanUp() {
                 log.error("Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
             }
         }
-
+        if(testContext.getScenarioContext().getSessionEventForwardingRegisters() != null) {
+            for (SessionEventForwardingRegister sessionEventForwardingRegister : testContext.getScenarioContext().getSessionEventForwardingRegisters()) {
+                try {
+                    sessionEventForwardingRegisterRepository.delete(sessionEventForwardingRegister);
+                } catch (DataIntegrityViolationException e) {
+                    log.error("Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
+                }
+            }
+        }
```

Comment:
As a side note. All this duplication is unnecessary. It's fine to have this clean up code in just one of the steps classes and it get run for all. I keep meaning to refactor this.